src/main/java/org/sunbird/trending/controller/TrendingServiceImpl.java

```diff
@@ -178,84 +179,72 @@ private String getEndDateFormat() {
+           if (request.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Request is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           HashMap<String, Object> filter = request.get(FILTERS) == null ? new HashMap<>() : ((HashMap<String, Object>) request.get(FILTERS));
+           if (filter.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Filter is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           List<String> contextTypeList = new ArrayList<>();
+           if (CollectionUtils.isNotEmpty(contextTypeList)) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "ContextType is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           String org = (filter).get(ORGANISATION) == null ? "" : ((String) (filter).get(ORGANISATION));
+           if (org.isEmpty()) {
+               response.getParams().setStatus(Constants.FAILED);
+               response.put(MESSAGE, "Organization is Missing");
+               response.setResponseCode(HttpStatus.BAD_REQUEST);
+               return response;
+           }
+           Map<String, String> payloadToRedisKeyMapping = serverProperties.getPayloadToRedisKeyMapping();
+           contextTypeList.addAll(payloadToRedisKeyMapping.keySet());
+           Map<String, Object> aggregateData = new HashMap<>();
+           List<String> topContentRedis;
+           List<String> limitCourses;
+           int limit = Optional.ofNullable(request.get(LIMIT)).map(l -> (Integer) l).orElse(0);
+           for (String contextType : contextTypeList) {
+               String contextTypeValue = payloadToRedisKeyMapping.get(contextType);
+               topContentRedis = redisCacheMgr.hget(contextTypeValue, serverProperties.getRedisInsightIndex(), new String[]{org});
+               limitCourses = this.fetchIds(topContentRedis.get(0), limit, "");
+               if (CollectionUtils.isNotEmpty(limitCourses) && null != limitCourses.get(0)) {
+                   aggregateData.put(contextType, limitCourses);
+               }
+           }
+           Map<String, Object> compositeSearchRes;
+           List<Map<String, Object>> contentList = new ArrayList<>();
+           Map<String, Object> resultMap;
+           if (!aggregateData.isEmpty()) {
+               List<String> compositeKeyList = new ArrayList<>(payloadToRedisKeyMapping.keySet());
+               for (int i = 0; i < aggregateData.size(); i++) {
+                   List<String> searchIds = (List<String>) aggregateData.get(compositeKeyList.get(i));
```

Comment:
Why you do composite search contentService.readContentFromCache for getting the value for content and add the same way i add for topReview.

Reply:
Implimented as suggested.